lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -683,9 +685,7 @@ def scrape_identifier(self):
         """Format the topology information into a scrape identifier."""
         # This is used only by Metrics[Consumer|Provider] and does not need a
         # unit name, so only check for the charm name
-        return "juju_{}_prometheus_scrape".format(
-            "_".join([self.model, self.model_uuid[:7], self.application, self.charm_name])  # type: ignore
-        )
+        return "juju_{}_prometheus_scrape".format(self.identifier)
```

Comment:
This looks significantly different. Is this a breaking change?

Reply:
@sed-i if you are asking this question, we may not have sufficient unit tests. Care to validate that?

Reply:
I believe it does not truncate the UUID. @mmanciop would you like to continue truncating the UUID? Otherwise the change should not break anything that I am aware of.

Reply:
If we truncate the id, we need to use a different matcher (~=) in promql-transform

Reply:
I am actually curious to know whether truncating the model_uuid would break the itests. @simskij ?

Reply:
@mmanciop I believe this only affects the file name.

Reply:
I think the right think to do here is eliminate the `scrape_identifier()` method using only the `topology.identifier()` method and ensuring that it truncates UUID.

Reply:
I am resolving this discussion so we can move conversation to https://github.com/canonical/prometheus-k8s-operator/issues/240

Reply:
well, they were truncated prior to the change, and with this change they no longer are. I expect us to keep the current behavior (ie. truncate at 8 chars). But also, as Michele wrote, this should break tests, or we are clearly missing some vital tests, no?